lib/debug_logging/class_notifier.rb

```diff
@@ -20,29 +21,39 @@ def notifies(*methods_to_notify)
             config_proxy = if (proxy = instance_variable_get(DebugLogging::Configuration.config_pointer('k', method_to_notify)))
                              proxy
                            else
-                             proxy = if !payload.empty?
-                                       Configuration.new(**debug_config.to_hash.merge(payload))
+                             proxy = if !config_opts.empty?
+                                       DebugLogging::Configuration.new(**debug_config.to_hash.merge(config_opts))
                                      else
                                        debug_config
                                      end
                              proxy.register(method_to_notify)
                              instance_variable_set(DebugLogging::Configuration.config_pointer('k', method_to_notify), proxy)
+                             ActiveSupport::Notifications.subscribe(
+                               DebugLogging::ArgumentPrinter.debug_event_name_to_s(method_to_notify: method_to_notify)
+                             ) do |*debug_args|
+                               proxy.log do
+                                 DebugLogging::LogSubscriber.log_event(ActiveSupport::Notifications::Event.new(*debug_args))
+                               end
+                             end
                              proxy
                            end
             ActiveSupport::Notifications.instrument(
-              debug_event_name_to_s(method_to_notify: method_to_notify), { debug_args: args }.merge(payload)
+              DebugLogging::ArgumentPrinter.debug_event_name_to_s(
+                method_to_notify: method_to_notify),
+              {
+                debug_args: args,
+                payload: payload,
+                config_proxy: config_proxy
+              }
             ) do
-              original_method.call(*args, &block)
+              if args.size == 1 && (harsh = args[0]) && harsh.is_a?(Hash)
```

Comment:
This resolves the Ruby 2.7 *warning* and Ruby 3.0 *error* about using hashes as named arguments.